code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1892,6 +1893,96 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 				H.adjustBruteLoss(LOW_PRESSURE_DAMAGE * H.physiology.pressure_mod * delta_time)
 				H.throw_alert("pressure", /atom/movable/screen/alert/lowpressure, 2)
 
+/datum/species/proc/handle_gas_interaction(mob/living/carbon/human/humi, datum/gas_mixture/environment, delta_time, times_fired)
+	if((humi?.wear_suit?.clothing_flags & STOPSPRESSUREDAMAGE) && (humi?.head?.clothing_flags & STOPSPRESSUREDAMAGE))
+		return
+
+	for(var/gas_id in environment.gases)
+		var/gas_amount = environment.gases[gas_id][MOLES]
+		switch(gas_id)
+			if(/datum/gas/antinoblium)
+				if(gas_amount > 30)
+					humi.apply_status_effect(/datum/status_effect/corrosion_curse)
+			if(/datum/gas/freon)
+				if(isplasmaman(humi))
+					if(gas_amount > 5 && gas_amount < 10)
+						to_chat(humi, span_warning(pick("Your bones feels hot.", "You feel hot.")))
+					else if(gas_amount >= 10)
+						to_chat(humi, span_warning(pick("Your bones starts to heat up!", "You feel very hot!")))
+						humi.adjustStaminaLoss(- gas_amount * 0.15 * delta_time)
+						humi.adjust_bodytemperature(max(gas_amount * 0.25, BODYTEMP_HEATING_MAX) * delta_time)
+					continue
+				if(gas_amount > 5 && gas_amount < 10)
+					to_chat(humi, span_warning(pick("Your skin feels cold.", "You feel cold.")))
+				else if(gas_amount >= 10)
+					to_chat(humi, span_warning(pick("Your skin starts to freeze!", "You feel very cold!")))
+					var/current_stamina = humi.getStaminaLoss()
+					if(current_stamina > 25)
+						humi.adjustStaminaLoss(gas_amount * 0.15 * delta_time)
+					humi.adjust_bodytemperature(max(- gas_amount * 0.25, BODYTEMP_COOLING_MAX) * delta_time)
+					if(humi.bodytemperature < 265)
+						humi.apply_status_effect(/datum/status_effect/freon)
+			if(/datum/gas/halon)
+				if(gas_amount > 15)
+					humi.apply_status_effect(/datum/status_effect/stoned)
+			if(/datum/gas/healium)
```

Comment:
a gas that heals you, works better if breathed but sleeps you

Reply:
im really iffy on this, having a dedicated healing gas is already kind of iffy but removing its one downside in exchange for less healing (via skin i mean) is a bad direction. either its too slow to matter in any practical sense in which case why have it or its free healing if you have enough of it, can you explain why this would add depth to the system?

Reply:
same tbh, the healing is not too strong, but this also heals dead people (not wounds) so there is a usecase for it

Reply:
Why does it add depth though? Dead healing is a negative not a plus I think, since that being medical's area is important. I agree with kyler totally